src/Illuminate/Validation/Validator.php

```diff
@@ -385,17 +385,21 @@ protected function validateAttribute($attribute, $rule)
         // If we have made it this far we will make sure the attribute is validatable and if it is
         // we will call the validation method with the attribute. If a method returns false the
         // attribute is invalid and we will add a failure message for this failing attribute.
-        $validatable = $this->isValidatable($rule, $attribute, $value);
+        if (! $this->isValidatable($rule, $attribute, $value)) {
+            return;
+        }
+
+        if (is_a($rule, RuleContract::class, true)) {
+            if (is_string($rule)) {
+                $rule = app($rule, $this->parseNamedParameters($parameters));
```

Comment:
We can't use the global app helper here like this. We'd need to inject the container as wel into this class and frankly I don't know if that's worth the effort.

Reply:
Right - I thought this was bad code when I wrote it but wasn't sure of the best way of relying on the container here.

Providing the container to the validators from the factory seems like a strange architecture.

Reply:
Oh, apparently validators already have the container:
https://github.com/laravel/framework/blob/f25076bd663b2891b6216c769274e12e239e44ab/src/Illuminate/Validation/Validator.php#L35

Reply:
Ok I've pushed a commit that uses the inbuilt container rather than the `app` global.